tests/inputs/example_service/test_example_service.py

```diff
@@ -0,0 +1,108 @@
+    ExampleRequest,
+    ExampleResponse,
+)
+
+
+class ExampleService(ExampleServiceImplementation):
+    async def example_unary_unary(
+        self, example_string: str, example_integer: int
+    ) -> "ExampleResponse":
+        return ExampleResponse(
+            example_string=example_string,
+            example_integer=example_integer,
+        )
+
+    async def example_unary_stream(
+        self, example_string: str, example_integer: int
+    ) -> AsyncIterator["ExampleResponse"]:
+        response = ExampleResponse(
+            example_string=example_string,
+            example_integer=example_integer,
+        )
+        yield response
+        yield response
+        yield response
+
+    async def example_stream_unary(
+        self, example_request_iterator: AsyncIterable["ExampleRequest"]
+    ) -> "ExampleResponse":
+        async for example_request in example_request_iterator:
+            return ExampleResponse(
+                example_string=example_request.example_string,
+                example_integer=example_request.example_integer,
+            )
+
+    async def example_stream_stream(
+        self, example_request_iterator: AsyncIterable["ExampleRequest"]
+    ) -> AsyncIterator["ExampleResponse"]:
+        async for example_request in example_request_iterator:
+            yield ExampleResponse(
+                example_string=example_request.example_string,
+                example_integer=example_request.example_integer,
+            )
+
+
+async def async_test_server_start():
+    host = "127.0.0.1"
+    port = 13337
+
+    test_string = "test string"
+    test_int = 42
+
+    # start server
+    server = Server([ExampleService()])
+    await server.start(host, port)
+
+    # start client
+    channel = Channel(host=host, port=port)
+    stub = ExampleServiceStub(channel)
+
+    # unary unary
+    response = await stub.example_unary_unary(
+        example_string="test string",
+        example_integer=42,
+    )
+    assert response.example_string == test_string
+    assert response.example_integer == test_int
+
+    # unary stream
+    async for response in stub.example_unary_stream(
+        example_string="test string",
+        example_integer=42,
+    ):
+        assert response.example_string == test_string
+        assert response.example_integer == test_int
+
+    # stream unary
+    request = ExampleRequest(
+        example_string=test_string,
+        example_integer=42,
+    )
+
+    async def request_iterator():
+        yield request
+        yield request
+        yield request
+
+    response = await stub.example_stream_unary(request_iterator())
+    assert response.example_string == test_string
+    assert response.example_integer == test_int
+
+    # stream stream
+    async for response in stub.example_stream_stream(request_iterator()):
+        assert response.example_string == test_string
+        assert response.example_integer == test_int
+
+
+def test_server_start():
```

Comment:
I don't think it should be necessary to run your own event loop. Did you try use `@pytest.mark.asyncio`? 

example: https://github.com/danielgtaylor/python-betterproto/blob/master/tests/grpc/test_grpclib_client.py#L174

Reply:
I didn't know that was a thing. Thanks for the pointer!